packages/block-library/src/button/edit.js

```diff
@@ -140,8 +140,10 @@ function ButtonEdit( {
 					}
 				) }
 				style={ {
-					backgroundColor: ! customGradient && backgroundColor.color,
-					background: customGradient,
+					...( customGradient ?
+						{ background: customGradient } :
+						( { backgroundColor: backgroundColor.color } )
```

Comment:
I'm not sure if the extra parentheses were intentional, they don't seem to be applied to the above line, but it might be nice to make it consistent one way or the other.

```suggestion
						{ backgroundColor: backgroundColor.color }
```